src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -718,34 +904,63 @@ private void Write(byte[] bytes)
             binaryWriter.Write(bytes);
         }
 
+        private void Write(byte b)
+        {
+            binaryWriter.Write(b);
+        }
+
         private void Write(bool boolean)
         {
             binaryWriter.Write(boolean);
         }
 
-        private void Write(string text)
+        private void WriteDeduplicatedString(string text)
```

Comment:
Technically I could rename Write to WriteRaw (it's only called once) and rename WriteDeduplicatedString back to Write to minimize the diff. But in about half the cases there'll still be a diff because half the cases call WriteOptionalString.

Reply:
WriteOptionalString is no longer needed because WriteDeduplicatedString has the ability to write null by just using reserved index 0.

Reply:
The historical reason is I first wasn't sure whether deduping all strings would be worse or better, so I deduped just some of them. But gradually I deduped more and more and it got better and better, so deduping all ended up being optimal. Of course I didn't know that from the outset so I kept a parallel name all the way.

Reply:
Thanks for explaining!